### DISCUSSION TITLE
rule to trigger and send telemetry value as a data item to a Domoticz IDX "Custom Sensor"

### DISCUSSION THREAD

**Discussion by K1WIZ** [arendst/Tasmota#10768] [Q&A] [answered]
Hi,

Looking for help on how to send a telemetry value as a custom data item to a Domoticz dummy custom sensor IDX.

I have:
tele/tasmota_E46F3E/STATE = {"Time":"2021-01-30T17:36:25","Uptime":"0T00:05:09","UptimeSec":309,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"MIND CONTROL BG","BSSId":"FC:EC:DA:38:D5:8A","Channel":11,"RSSI":50,"Signal":-75,"LinkCount":1,"Downtime":"0T00:00:03"}}

I want to get the value of "UptimeSec" and send that number to an IDX for a custom sensor item on my Domoticz dashboard.
I tried creating a rule for it, but maybe I didn't do it correctly. Looking for help with sending this value:

rule1 on Tele-STATE#UptimeSec do publish domoticz/in {"idx":53, "nvalue":0,"svalue":"%value%"} endon

It seems to take the rule from the console, but is not triggering and sending the iteger value for the "UptimeSec" value of 309 in this example.

Appreciate any help, I feel I'm on the right track, but may just be slightly off.

**Comment by sfromis** [comment, marked as answer]
Topic is not part of the JSON parsing specs for triggers. Try `on Tele-UptimeSec do ....`

**Reply by K1WIZ** [reply]
I did try this before I saw your reply:
rule1 on time#minute|3 do publish domoticz/in {"idx":53, "nvalue":0,"svalue":"%uptime%"} endon
this works great and sends the uptime in minutes to the custom sensor item on the domoticz dashboard.   Works great and now I have a graph showing an increasing uptime trend.

The application I'm using this for is detecting when a GFCI outlet trips on an HVAC condensate pump.   simple script compares new value to previous value and if not greater, Domoticz sounds an alarm.   I'm using a cheap wemos D1 mini that is plugged into the GFCI outlet to detect the trip.   Works pisser.   Thanks for your help.
![gfci-uptime](https://user-images.githubusercontent.com/25913189/106375655-ef20a580-635b-11eb-8438-7d5649284d1a.png)

**Reply by barbudor** [reply]
Sounds like a bit complicated while a rule `on system#boot` or `on power#boot` could directly publish a warning message on reboot.
You can also catch in your MQTT backend messages such as "INFO1/2/3" which are automatically publish at boot time

**Comment by K1WIZ** [comment]
but doing it that way would not enable me to graph the uptime in Domoticz
as well.   This way I acheive both.

On Sun, Jan 31, 2021 at 04:25 Barbudor <notifications@github.com> wrote:

> Sounds like a bit complicated while a rule on system#boot or on power#boot
> could directly publish a warning message on reboot.
> You can also catch in your MQTT backend messages such as "INFO1/2/3" which
> are automatically publish at boot time
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/arendst/Tasmota/discussions/10768#discussioncomment-324695>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGFWOZJXZUW7NWXI57Y6WZDS4UOZDANCNFSM4W2V7BSQ>
> .
>
-- 
Sent from my iPhone

**Comment by arendst** [comment]
And the rule can be shorted to ``on time#minute|3 do dzsend1 53,%uptime% endon``

knowing these command syntaxis:
```
  // DzSend1 <idx>,<values> - {\"idx\":<idx>,\"nvalue\":0,\"svalue\":\"<values>\",\"Battery\":xx,\"RSSI\":yy}
  // DzSend2 <idx>,<values> - USE_SHUTTER only - {\"idx\":<idx>,\"nvalue\":<position>,\"svalue\":\"<values>\",\"Battery\":xx,\"RSSI\":yy}
  // DzSend3 <idx>,<values> - {\"idx\":<idx>,\"nvalue\":<values>,\"Battery\":xx,\"RSSI\":yy}
  // DzSend4 <idx>,<state>  - {\"command\":\"switchlight\",\"idx\":<idx>,\"switchcmd\":\"<state>\"}
  // DzSend5 <idx>,<state>  - {\"command\":\"switchscene\",\"idx\":<idx>,\"switchcmd\":\"<state>\"}
```
